modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/SocialMediaIconsListFormatter.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\link\Plugin\Field\FieldFormatter\LinkFormatter;
+
+/**
+ * Displays typed links as a list of social icons.
+ *
+ * This formatter assumes that link categories will be compatible with
+ * media service names used in the "Social media links: horizontal" pattern.
+ *
+ * @see templates/patterns/social_media_links/social_media_links_horizontal.ui_patterns.yml
+ *
+ * @FieldFormatter(
+ *   id = "oe_theme_helper_social_media_icons_list_formatter",
+ *   label = @Translation("Social media icons list"),
+ *   description = @Translation("Display typed links as list of social icons."),
+ *   field_types = {
+ *     "typed_link"
+ *   }
+ * )
+ */
+class SocialMediaIconsListFormatter extends LinkFormatter {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    if (!$items->count()) {
+      return [];
+    }
+
+    $list_items = [];
+    $elements = parent::viewElements($items, $langcode);
+    foreach ($items as $delta => $item) {
+      $pattern = [
```

Comment:
Currently heigh of icon in the social_icon pattern is on 4px bigger than on the mockup. Is it important? If yes, then we should implement custom template for it because there isn't way to set icon size in social_icon pattern.
<img width="1177" alt="Screenshot 2020-07-24 at 16 33 36" src="https://user-images.githubusercontent.com/1183951/88397052-13f3f900-cdcc-11ea-849d-00daaead0c07.png">

Reply:
That's ok, no worries.

Reply:
Antonio said that current behaviour is acceptable.